src/SDK/MonitoringApplianceEmulator/ViewModels/ChartPropertyControlViewModel.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Azure.Monitoring.SmartDetectors.MonitoringApplianceEmulator.ViewModels
+{
+    using System;
+    using System.Globalization;
+    using System.Windows.Media;
+    using LiveCharts;
+    using LiveCharts.Configurations;
+    using LiveCharts.Wpf;
+    using Microsoft.Azure.Monitoring.SmartDetectors.MonitoringApplianceEmulator.Models.Chart;
+    using Microsoft.Azure.Monitoring.SmartDetectors.RuntimeEnvironment.Contracts;
+
+    /// <summary>
+    /// The view model class for the <see cref="ChartPropertyControlViewModel"/> control.
+    /// </summary>
+    public class ChartPropertyControlViewModel : ObservableObject
+    {
+        private static readonly Brush SeriesColor = Brushes.DeepSkyBlue;
+
+        private SeriesCollection seriesCollection;
+
+        private Func<double, string> xAxisFormatter;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChartPropertyControlViewModel"/> class.
+        /// </summary>
+        /// <param name="chartAlertProperty">The chart alert property that should be displayed.</param>
+        public ChartPropertyControlViewModel(ChartAlertProperty chartAlertProperty)
+        {
+            var chartValues = new ChartValues<DateTimeDataPoint>();
+
+            foreach (var dataPoint in chartAlertProperty.DataPoints)
+            {
+                DateTime dateTime;
+                try
+                {
+                    dateTime = Convert.ToDateTime(dataPoint.X, CultureInfo.InvariantCulture);
+                }
+                catch (Exception e) when (e is FormatException || e is InvalidCastException)
+                {
+                    throw new InvalidCastException($"The data point's Y value '{dataPoint.X}' is not of DateTime type", e);
+                }
+
+                double value;
+                try
+                {
+                    value = Convert.ToDouble(dataPoint.Y, CultureInfo.InvariantCulture);
+                }
+                catch (Exception e) when (e is FormatException || e is InvalidCastException || e is OverflowException)
+                {
+                    throw new InvalidCastException($"The data point's X value '{dataPoint.X}' is not of a numeric type", e);
+                }
+
+                chartValues.Add(new DateTimeDataPoint(dateTime, value));
+            }
+
+            /*
+             * Relevant for Column Series:
+             * Since we are using DateTime.Ticks as X, the width of the bar is 1 tick and 1 tick is 1 millisecond.
+             * In order to make our bars visible we need to change the unit of the chart. For the initial view are going to use hours.
+             * There is a future task (#1380564) to create this scale dynamically according to the X values range size.
```

Comment:
>There is a future task (#1380564) to create this scale dynamically according to the X values range size. [](start = 15, length = 104)

I don't think that'll be good enough, not even as a temporary solution. Can it be reduced to 5m instead of 1h? If not, 15m?

Reply:
yes, it can be reduced or increased to any size we want, do you already know what what will the X range size for CAD charts? will it be the same for all?

---
In reply to: [230770311](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/17#discussion_r230770311) [](ancestors = 230770311)

Reply:
On second thought - just take the difference between the first 2 points in the series, and use that as the timespan.

---
In reply to: [230770311](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/17#discussion_r230770311) [](ancestors = 230770311)

Reply:
or use (last - first)/total points

---
In reply to: [230771921](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/17#discussion_r230771921) [](ancestors = 230771921,230770311)